root/etc/e-smith/events/actions/nethserver-nextcloud-restore-rh-mariadb105-database

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+#
+# Copyright (C) 2021 Nethesis S.r.l.
+# http://www.nethesis.it - nethserver@nethesis.it
+#
+# This script is part of NethServer.
+#
+# NethServer is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or any later version.
+#
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+source /opt/rh/rh-mariadb105/enable
+
+if [[ ! -f '/var/lib/nethserver/backup/rh-mariadb105/nextcloud.sql' ]]; then
+    echo "[ERROR] Error the nextcloud SCL mariadb105 dump does not exist"
+    exit 1
+fi
+
+password=`perl -e "use NethServer::Password; print NethServer::Password::store('nextcloud');"`
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "CREATE DATABASE IF NOT EXISTS nextcloud;"
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock --database nextcloud < /var/lib/nethserver/backup/rh-mariadb105/nextcloud.sql
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "ALTER DATABASE nextcloud CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;"
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "grant all on nextcloud.* to 'nextcloud'@'localhost' identified by '$password';"
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "FLUSH PRIVILEGES"
+
+if [[ $? -ne 0 ]]; then
```

Comment:
This condition checks only the last (FLUSH PRIVILEGES) command exit code. 

- Use  the Bash `set -e` option to automatically exit as soon as an error occurs, OR
- Group together the SQL command strings in a single SQL stream and pipe it to a single  `mysql` command, i.e.:

      ( echo CMD1 ; echo CMD2 ; cat nextcloud.sql ; echo CMD3 ... ) | mysql

Reply:
hello it should be fixed